src/storage.rs

```diff
@@ -12,6 +46,115 @@ impl Default for FileStorage {
     }
 }
 
-pub trait Storage {}
+pub trait Storage {
+    fn get_username(&self) -> String;
+    fn get_password(&self) -> String;
+    fn get_token(&self) -> String;
+    fn get_course(&self) -> String;
+    fn get_task(&self) -> String;
+    fn get_language(&self) -> String;
+    fn get_option(&self) -> String;
+    fn get_file(&self) -> String;
+    fn set_username(&mut self, val: String);
+    fn set_password(&mut self, val: String);
+    fn set_token(&mut self, val: String);
+    fn set_course(&mut self, val: String);
+    fn set_task(&mut self, val: String);
+    fn set_language(&mut self, val: String);
+    fn set_option(&mut self, val: String);
+    fn set_file(&mut self, val: String);
+    fn save(&self);
```

Comment:
In this implementation, `save` and `delete` don't necessarily need to take a mutable reference to self (`&mut self`), but in tests that might be necessary. So I'd make it mutable anyways.